src/clima_diagnostics.jl

```diff
@@ -186,13 +233,60 @@ function orchestrate_diagnostics(
         active_compute[diag_index] || continue
         diag = scheduled_diagnostics[diag_index]
 
-        diag.variable.compute!(
+        diagnostic_handler.counters[diag_index] += 1
+
+        # We have three cases:
+        #
+        # 1. The variable has an in-place `compute!` function. In this case, we simply
+        # evaluate it and overwrite the associated storage.
+        #
+        # 2a. The variable has a `compute` function that returns a Field. In this case, we
+        # copy it over to the storage.
+        #
+        # 2b. The variable has a `compute` function that returns
+        # `Base.Broadcast.Broadcasted`. In this case, we update the value in our dictionary
+        # and materialize the expression in the next loop.
+
+        has_inplace_compute = !isnothing(diag.variable.compute!)
+        if has_inplace_compute
+            # Case 1
+            diag.variable.compute!(
+                diagnostic_handler.storage[diag_index],
+                integrator.u,
+                integrator.p,
+                integrator.t,
+            )
+        else
+            out_or_broadcasted =
+                diag.variable.compute(integrator.u, integrator.p, integrator.t)
+
+            if out_or_broadcasted isa Base.Broadcast.Broadcasted
+                # Case 2b
+                diagnostic_handler.broadcasted_expressions[diag] =
+                    out_or_broadcasted
+            else
+                # Case 2a
+                diagnostic_handler.storage[diag_index] .= out_or_broadcasted
+            end
```

Comment:
```suggestion
            Base.copyto!(diagnostic_handler.storage[diag_index],  out_or_broadcasted)
```
Both `Base.copyto!(::Field, ::Field)` and `Base.copyto!(::Field, ::Base.Broadcasted)` are supported